prometheus-metrics/src/main/scala/org/http4s/metrics/prometheus/Prometheus.scala

```diff
@@ -292,6 +336,18 @@ object Prometheus {
   // https://github.com/prometheus/client_java/blob/parent-0.6.0/simpleclient/src/main/java/io/prometheus/client/Histogram.java#L73
   private val DefaultHistogramBuckets: NonEmptyList[Double] =
     NonEmptyList(.005, List(.01, .025, .05, .075, .1, .25, .5, .75, 1, 2.5, 5, 7.5, 10))
+
+  // Prometheus expects exemplars as alternating key-value strings: k1, v1, k2, v2, ...
+  private def toFlatArray(m: Map[String, String]): Array[String] = {
```

Comment:
Is it worth doing any validation here, or just let the underlying library handle it? The k/v string can only be 128 charachters long

Reply:
Good catch. I'm unsure on the best way. Currently, providing overlong values will result in an error that we don't catch and hence fail entirely. That's not cool.

On the other hand, I'd like to avoid hardcoding any restrictions. Prometheus' simpleclient 1.0 was released last week and it seems to remove the 128 chars constraint, at least I can't find it in sources anymore. Also makes sense as they added an OTEL exporter where it probably doesn't apply anymore. They also introduced a new `Labels` class that replaces the alternating array parameter.

This leaves us with the following options:
1. Just document that the user is responsible to ensure validity, keeping crashing if they fail to do so. As I expect most users to use something safe like prometheus4cats, this might not happen often.
2. Validate and don't record an exemplar if it's invalid. We could use `new Exemplar(...)` from simpleclient 0.16.0 so it becomes visible and can be reconsidered when we upgrade to Prometheus simpleclient 1.0 one day. Could be opt-in/opt-out as it adds CPU consumption (validation will be run twice due to the limited API). Has the benefit of checking other things on the way, like invalid label names.
3. Update to simpleclient 1.0 right now. Probably not a good idea as bincompat seems completely broken and it's very fresh. Mid-term might warrant a dedicated module so we can cater both versions for a while.

Any preference between 1. and 2. from your side? Anything I missed? @janstenpickle

Reply:
I think 1) would probably be my preferance at this point - this isn't doing anything magic like pulling the exemplar out of a threadlocal, so the developer is entirely in control of being able to resolve any exceptions from the client. It might be nice to wrap the exception from the simpleclient if it's error message is in any way cryptic.

Reply:
Regarding 3) - let's discuss below